fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/Reference.java

```diff
@@ -223,6 +223,10 @@ public void insertUnchecked(@Nonnull final RelationalExpression newValue) {
     public void insertUnchecked(@Nonnull final RelationalExpression newValue, @Nullable final Map<PlanProperty<?>, ?> precomputedPropertiesMap) {
         // Call debugger hook to potentially register this new expression.
         Debugger.registerExpression(newValue);
+
+        Debugger.sanityCheck(() -> Verify.verify(members.isEmpty() ||
+                getResultType().equals(newValue.getResultType())));
```

Comment:
Is leaving this sanity check here ok or is it costly?

Reply:
`Debugger.sanityCheck()` only runs when a debugger is installed. It won't run in performance-sensitive environments.